scripts/cli/handler/workspace.py

```diff
@@ -0,0 +1,222 @@
+import os
+import termios
+import shutil
+import signal
+import subprocess
+import sys
+import tarfile
+import json
+import copy
+
+from os.path import join
+
+from util.configfilehandler import ConfigFileHandler
+from util import util
+
+def __list_workspace(config_file_handler):
+  '''Lists the available workspaces.'''
+  print '\n'.join(config_file_handler.names())
+
+def kill_child_processes(parent_pid, sig=signal.SIGTERM):
+  ps_command = subprocess.Popen("ps -o pid --ppid %d --noheaders" % parent_pid, shell=True, stdout=subprocess.PIPE)
+  ps_output = ps_command.stdout.read()
+  retcode = ps_command.wait()
+  assert retcode == 0, "ps command returned %d" % retcode
+  for pid_str in ps_output.split("\n")[:-1]:
+    os.kill(int(pid_str), sig)
+
+def __delete_workspace(config_file_handler, args):
+  '''Deletes a workspace (project dir, SQL database, CodeChecker runs)'''
+  if not config_file_handler.exists(args.delete):
+    sys.exit("Workspace '" + args.delete + "' does not exist!")
+
+  #--- Remove from database ---#
+
+  projectdir = join(args.workdir, args.delete)
+  success = util.pgsql_cmd(projectdir, args, 'dropdb', args.delete)
+
+  if (success != 0):
+    sys.exit("Couldn't delete database " + args.delete + " because it doesn't "
+    + "exist or bad connection (Host: " + args.dbhost + " Port: "
+    + str(args.dbport) + " User: " + args.dbuser + ")")
+
+  #--- Remove the project directory ---#
+
+  if os.path.exists(projectdir):
+    shutil.rmtree(projectdir)
+ 
+  #--- Remove from config file ---#
+
+  config_file_handler.remove(args.delete)
+  config_file_handler.save()
+
+def __rename_workspace(config_file_handler, args):
+  '''Renames a workspace (project dir, SQL database). CodeChecker run name will
+  not be modified.'''
+
+  if not config_file_handler.exists(args.rename[0]):
+    sys.exit("Workspace '" + args.rename[0] + "' does not exists!")
+  if config_file_handler.exists(args.rename[1]):
+    sys.exit("Workspace '" + args.rename[0] + "' already exists!")
+
+  #--- Rename in config file ---#
+
+  config_file_handler.rename(args.rename[0], args.rename[1])
+  config_file_handler.save()
+
+  #--- Rename the project directory ---#
+
+  projectdir_from = join(args.workdir, args.rename[0])
+  projectdir_to   = join(args.workdir, args.rename[1])
+
+  if os.path.exists(projectdir_from):
```

Comment:
`config_file_handler.exists` function checks if the name parameter is an existing workspace. It checks the existence in the workspace file, not in the system.